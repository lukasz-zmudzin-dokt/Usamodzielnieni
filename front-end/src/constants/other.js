const DEFAULT_INPUT = "-- Wybierz --";

export { DEFAULT_INPUT };
