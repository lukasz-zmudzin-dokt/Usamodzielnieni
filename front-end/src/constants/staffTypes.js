export const staffTypes = {
  VERIFICATION: "staff_verification",
  CV: "staff_cv",
  JOBS: "staff_jobs",
  BLOG_CREATOR: "staff_blog_creator",
  BLOG_MODERATOR: "staff_blog_moderator",
};
