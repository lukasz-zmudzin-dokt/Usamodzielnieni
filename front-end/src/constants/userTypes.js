export const userTypes = {
  STANDARD: "standard",
  STAFF: "staff",
  EMPLOYER: "employer",
};
