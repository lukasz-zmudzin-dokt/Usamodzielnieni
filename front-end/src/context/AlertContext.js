import React, { useState, createContext } from "react";

export const AlertContext = createContext({
  open: false,
  message: "",
  setOpen: () => {},
});

export const AlertProvider = (props) => {
  const [open, setOpen] = useState(false);
  const [message, setMessage] = useState("");
  const [variant, setVariant] = useState("danger") // success
  const data = {
    open,
    message,
    variant,
    changeVisibility: (newOpen) => setOpen(newOpen),
    changeMessage: (newMessage,newVariant = variant) => {
      setOpen(false);
      setMessage(newMessage);
      setVariant(newVariant);
    },
    showAlert:  (newMessage,newVariant) => {
      data.changeMessage(newMessage,newVariant);
      data.changeVisibility(true);
    }
  };
  return <AlertContext.Provider value={data} {...props} />;
};
