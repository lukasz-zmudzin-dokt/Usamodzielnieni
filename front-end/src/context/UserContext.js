import React, { useState } from "react";
import Cookies from "universal-cookie";

export const UserContext = React.createContext({
  type: undefined, // Employer,Standard,Staff
  token: undefined,
  login: () => {},
  logout: () => {}
});
const cookies = new Cookies();

export const UserProvider = props => {
  const [type, setType] = useState(cookies.get("type"));
  const [token, setToken] = useState(cookies.get("token"));
  const user = {
    token,
    type,
    login: (newToken, newType) => {
      cookies.set("token", newToken, { path: "/" });
      cookies.set("type", newType, { path: "/" });
      setToken(newToken);
      setType(newType);
    },
    logout: () => {
      cookies.remove("token", { path: "/" });
      cookies.remove("type", { path: "/" });
      setToken(undefined);
      setType(undefined);
    }
  };
  return <UserContext.Provider value={user} {...props} />;
};
