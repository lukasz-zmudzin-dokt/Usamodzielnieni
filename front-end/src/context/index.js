export { UserContext, UserProvider } from "context/UserContext";
export { AlertContext, AlertProvider } from "context/AlertContext";
export { ChatContext, ChatProvider } from "context/ChatContext";
export {
  NotificationsContext,
  NotificationsProvider,
} from "context/NotificationsContext";
