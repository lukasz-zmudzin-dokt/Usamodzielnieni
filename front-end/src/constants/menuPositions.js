import { userTypes } from "./userTypes";
import { staffTypes } from "./staffTypes";
import { paths } from "./paths";

export default [       
    {
        name: "Kreator CV",
        path: paths.CV_CREATOR,
        allowed: [userTypes.STANDARD],
        verified: true
    },
    {
        name: "Moje CV",
        path: paths.MY_CVS,
        allowed: [userTypes.STANDARD],
        verified: true
    },
    {
        name: "Akceptacja CV",
        path: paths.CV_APPROVAL,
        allowed: [staffTypes.CV],
        verified: true
    },
    {
        name: "Oferty pracy",
        path: paths.JOB_OFFERS,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER]
    },
    {
        name: "Dodaj ofertę",
        path: paths.OFFER_FORM,
        allowed: [userTypes.EMPLOYER],
        verified: true
    },
    {
        name: "Moje oferty",
        path: paths.MY_OFFERS,
        allowed: [userTypes.EMPLOYER],
        verified: true
    },
    {
        name: "Blogi",
        path: paths.BLOG_PAGE,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER]
    },
    {
        name: "Dodaj post",
        path: paths.BLOG_FORM,
        allowed: [staffTypes.BLOG_CREATOR],
        verified: true
    },
    {
        name: "Telefony",
        path: paths.CONTACT_PAGE,
        allowed: undefined, //wszyscy są upoważnieni - zalogowani i niezalogowani
    },
    {
        name: "Kroki",
        path: paths.STEPS,
        allowed: undefined //wszyscy są upoważnieni - zalogowani i niezalogowani
    },
    {
        name: "Chat",
        path: paths.CHATS,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER],
        verified: true,
    }
];
