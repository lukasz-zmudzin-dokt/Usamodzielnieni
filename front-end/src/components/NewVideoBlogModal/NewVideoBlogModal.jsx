import { Button, Form, Modal } from "react-bootstrap";
import { Redirect } from "react-router-dom";
import React, { useContext, useRef, useState } from "react";
import {
  postBlogPost,
  reserveSpace,
  uploadPhoto,
} from "Views/BlogPostForm/functions/apiCalls";
import {
  VIDEOBLOG_CATEGORY,
  VIDEOBLOG_INITIAL_CONTENT,
  VIDEOBLOG_TAGS,
} from "constants/videoBlogInitialValues";
import FormGroup from "../FormGroup";
import { AlertContext } from "context/AlertContext";
import { approveFileSize } from "../../utils/approveFile/approveFile";

const handleVideoBlogAddition = async (token, data, head) => {
  try {
    const { id } = await reserveSpace(token);
    await postBlogPost(data, token, "POST", id);
    head && (await uploadPhoto(id, head, token, "header"));
    return id;
  } catch (e) {
    throw e;
  }
};

const NewVideoBlogModal = ({ show, setShow, user }) => {
  const [newTitle, setNewTitle] = useState("");
  const [validated, setValidated] = useState(false);
  const [redirect, setRedirect] = useState(false);
  const [id, setId] = useState(null);
  const [label, setLabel] = useState("");

  const alertC = useRef(useContext(AlertContext));
  let fileInput = useRef(null);

  const clearInput = () => {
    setNewTitle("");
    setValidated(false);
    setShow(false);
  };

  const onChange = () => {
    const file = fileInput.current?.files?.[0];
    if (!approveFileSize(file)) {
      alertC.current.showAlert(
        "Wybrany plik jest za duży. Maksymanly rozmiar pliku to 15 MB."
      );
      fileInput = null;
    } else {
      setLabel(file.name);
    }
  };

  const onSubmit = async (e) => {
    e.preventDefault();
    setValidated(true);
    if (e.currentTarget.checkValidity() === false) {
      e.stopPropagation();
    } else {
      let data = {
        title: newTitle,
        category: VIDEOBLOG_CATEGORY,
        content: VIDEOBLOG_INITIAL_CONTENT,
        tags: VIDEOBLOG_TAGS,
      };
      let head = fileInput.current.files?.[0];
      try {
        const res = await handleVideoBlogAddition(user.token, data, head);
        setId(res);
        clearInput();
        alertC.current.showAlert(
          "Pomyślnie założono nowy wideoblog.",
          "success"
        );
        setRedirect(true);
      } catch (e) {
        alertC.current.showAlert("Wystąpił błąd przy dodawaniu wideobloga.");
      }
    }
  };

  return (
    <>
      <Modal show={show} onHide={(e) => clearInput()}>
        <Modal.Header closeButton>
          <Modal.Title>Nowy wideoblog</Modal.Title>
        </Modal.Header>
        <Form noValidate validated={validated} onSubmit={onSubmit}>
          <Modal.Body>
            <Form.Label>
              (Opcjonalnie) Dodaj zdjęcie w nagłówku bloga
            </Form.Label>
            <Form.File
              id="custom-file"
              ref={fileInput}
              custom
              onChange={onChange}
              label={label || "Dodaj zdjęcie..."}
              accept="image/*"
              data-browse="Wybierz plik"
              className="mb-2"
            />
            <FormGroup
              header="Tytuł bloga"
              id="newTitle"
              setVal={setNewTitle}
              required
              incorrect="Podaj tytuł wideobloga"
              val={newTitle}
            />
          </Modal.Body>
          <Modal.Footer>
            <Button variant="primary" type="success">
              Stwórz nowy wideoblog
            </Button>
          </Modal.Footer>
        </Form>
      </Modal>
      {redirect && <Redirect to={"/blog/blogpost/" + id} />}
    </>
  );
};

export default NewVideoBlogModal;
