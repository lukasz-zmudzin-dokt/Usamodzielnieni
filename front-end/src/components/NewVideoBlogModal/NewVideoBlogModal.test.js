import NewVideoBlogModal from "./NewVideoBlogModal";
import {staffTypes} from "constants/staffTypes";
import React from "react";
import {userTypes} from "constants/userTypes";
import {render, fireEvent, wait} from "@testing-library/react";
import {createMemoryHistory} from 'history';
import {AlertContext} from "context/AlertContext";
import {Router} from "react-router-dom";
import {UserContext} from "context/UserContext";

const renderWithRouter = (
    ui,
    contextA,
    {
        route = "/user",
        history = createMemoryHistory({ initialEntries: [route] }),
    } = {}
) => {
    let context = {
        token: 123,
        type: userTypes.STAFF,
        data: {
            group_type: [staffTypes.BLOG_CREATOR]
        }
    };
    return {
        ...render(
            <UserContext.Provider value={context}>
                    <Router history={history}>{ui}</Router>
            </UserContext.Provider>
        ),
        history,
    };
};

describe("blog modal test", () => {
    let submitFail, photoFail, reserveFail, user, show, setShow, alertC;
    beforeAll(() => {
        global.fetch = jest.fn().mockImplementation((input) => {
            return new Promise((resolve) => {
                if (submitFail || photoFail || reserveFail) {
                    resolve({status: 500})
                } else {
                    switch(input) {
                        case "submit":
                            resolve({status: 200, json: () => Promise.resolve("")});
                            break;
                        case "photo":
                            resolve({status: 200, json: () => Promise.resolve("")});
                            break;
                        default:
                            resolve({status: 201, json: () => Promise.resolve({id: 123123})});
                            break;
                    }
                }
            });
        });
    });

    beforeEach(() => {
        submitFail = false;
        photoFail = false;
        reserveFail = false;
        show = true;
        setShow = jest.fn();
        user = {
            token: 123,
            type: userTypes.STAFF,
            data: {
                group_type: [staffTypes.BLOG_CREATOR]
            }
        };
        alertC = {
            showAlert: jest.fn()
        }
    });

    it('should match snapshot', () => {
        const {getByRole} = render(
            <NewVideoBlogModal user={user} setShow={setShow} show={show} />
        );
        const container = getByRole("dialog");
        expect(container).toMatchSnapshot();
    });

    it('should validate form, render alert and redirect', async () => {
        const {history, getByText, getByLabelText} = renderWithRouter(
            <AlertContext.Provider value={alertC}>
                <NewVideoBlogModal user={user} setShow={setShow} show={show} />
            </AlertContext.Provider>
        );

        fireEvent.change(getByLabelText("Tytuł bloga"), {
            target: {value: "qweqwe"}
        });
        fireEvent.click(getByText("Stwórz nowy wideoblog"));

        await wait(() => expect(fetch).toHaveBeenCalledTimes(2));
        expect(alertC.showAlert).toHaveBeenCalledWith(
            "Pomyślnie założono nowy wideoblog.","success"
        );
        expect(history.location.pathname).toEqual("/blog/blogpost/123123")
    });

});