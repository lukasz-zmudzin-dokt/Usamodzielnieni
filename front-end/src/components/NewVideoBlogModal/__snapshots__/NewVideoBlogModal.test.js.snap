// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog modal test should match snapshot 1`] = `
<div
  aria-modal="true"
  class="fade modal show"
  role="dialog"
  style="display: block;"
  tabindex="-1"
>
  <div
    class="modal-dialog"
    role="document"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header"
      >
        <div
          class="modal-title h4"
        >
          Nowy wideoblog
        </div>
        <button
          class="close"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
          <span
            class="sr-only"
          >
            Close
          </span>
        </button>
      </div>
      <form
        class=""
        novalidate=""
      >
        <div
          class="modal-body"
        >
          <label
            class="form-label"
          >
            (Opcjonalnie) Dodaj zdjęcie w nagłówku bloga
          </label>
          <div
            class="mb-2 custom custom-file"
          >
            <input
              accept="image/*"
              class="custom-file-input"
              id="custom-file"
              type="file"
            />
            <label
              class="custom-file-label"
              data-browse="Wybierz plik"
              for="custom-file"
            >
              Dodaj zdjęcie...
            </label>
          </div>
          <div
            class="form-group"
          >
            <label
              class="form-label"
              for="newTitle"
            >
              Tytuł bloga
            </label>
            <input
              class="form-control"
              data-testid="default"
              id="newTitle"
              maxlength="50"
              minlength="1"
              placeholder="Tytuł bloga"
              required=""
              type="text"
              value=""
            />
            <div
              class="invalid-feedback"
            >
              Podaj tytuł wideobloga
            </div>
          </div>
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-primary"
            type="success"
          >
            Stwórz nowy wideoblog
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
